scg-system/src/main/java/io/telicent/core/SmartCacheGraph.java

```diff
@@ -119,4 +132,51 @@ private static boolean isAuthEnabled() {
     private static boolean isInitialCompactionEnabled() {
         return !Configurator.get(FMod_InitialCompaction.DISABLE_INITIAL_COMPACTION, Boolean::parseBoolean, false);
     }
+
+    private static String convertToAbsolutePathURI(String fileUri) {
+        // Remove the "file:" prefix to get the relative path
+        String relativePath = fileUri.substring(5);
+
+        // Convert the relative path to an absolute path
+        File file = new File(relativePath);
+        String absolutePath = file.getAbsolutePath();
+
+        // Return the absolute path as a file URI
+        return "file:/" + absolutePath;
+    }
+
+    private static void convertYamlConfigToRDF(String... args) {
+        YAMLConfigParser ycp = new YAMLConfigParser();
+        RDFConfigGenerator rcg = new RDFConfigGenerator();
+        String configPath;
+        String pattern = ".*\\.(yaml|yml)$";
+        Pattern regex = Pattern.compile(pattern);
+        for (int i = 0; i < args.length; i++) {
+            if (args[i].equalsIgnoreCase("--conf") && i + 1 < args.length ) {
```

Comment:
This feels a little brittle, looking at the Jena code this argument can be either `--config` or `--conf`, and it does allow for specifying the value inline e.g. `--conf=file`

You could also avoid the loop if you did something like:

```java
// Before the for loop
int confIndex = ArrayUtils.indexOf(args, "--config");
if (confIndex == -1) {
  confIndex = ArrayUtils.indexOf(args, "--conf");
}
// Instead of the for loop
if (confIndex != -1 && confIndex + 1 < args.length) {
  // Your existing logic from inside your for loop
}
```

This doesn't cope with the `--conf=file` form but for now I think that's fine as we're going to be controlling how this gets run in our deployments so we can guarantee we're always doing the `--conf file` form in our arguments

Reply:
Although in light on @robwtelicent's discussion thread about what happens if the user specifies multiple `--conf` options probably best to leave it as a `for` loop